src/components/term_activity/TermActivity.jsx

```diff
@@ -0,0 +1,259 @@
+import { vars } from "../../theme/variables";
+const { gray700, gray600, gray300 } = vars;
+
+const headCells = [
+    { id: 'label', label: 'Label' },
+    { id: 'organization', label: 'Organization' },
+    { id: 'type', label: 'Type' },
+    { id: 'description', label: 'Description' },
+    { id: 'timestamp', label: 'Timestamp' }
+];
+
+const descriptionTextStyle = {
+    color: gray600,
+    textOverflow: 'ellipsis',
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+    minWidth: '11rem',
+    maxWidth: '31.25rem'
+};
+
+const getChipColor = (type) => ({
+    Approved: 'success',
+    Edit: 'warning',
+    Rejected: 'error',
+    Imported: 'secondary',
+    'New term': 'primary'
+}[type]);
+
+
+const TermActivity = () => {
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+    const [rows, setRows] = React.useState([]);
+    const [order, setOrder] = React.useState('asc');
+    const [orderBy, setOrderBy] = React.useState('prefix');
+    const [selected, setSelected] = React.useState([]);
+    const [page, setPage] = React.useState(1);
+    const [numberOfVisiblePages, setNumberOfVisiblePages] = React.useState(0);
+    const [pageOptions, setPageOptions] = useState([]);
+
+
+    const sortedRows = React.useMemo(
+        () => {
+            const startIndex = (page - 1) * numberOfVisiblePages;
+            const endIndex = startIndex + numberOfVisiblePages;
+
+            return stableSort(rows, getComparator(order, orderBy)).slice(startIndex, endIndex);
+        },
+        [rows, order, orderBy, page, numberOfVisiblePages]
+    );
+
+    const handlePageChange = (event, value) => {
+        setPage(value);
+    };
+
+    const handleNumberOfPagesChange = (event) => {
+        setNumberOfVisiblePages(event.target.value);
+        setPage(1);
+    };
+
+    const handleSelectAllClick = (event) => {
+        if (event.target.checked) {
+            const newSelected = rows.map((n) => n.id);
+            setSelected(newSelected);
+            return;
+        }
+        setSelected([]);
+    };
+
+    const handleClick = (event, id) => {
+        const selectedIndex = selected.indexOf(id);
+        let newSelected = [];
+
+        if (selectedIndex === -1) {
+            newSelected = newSelected.concat(selected, id);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1),
+            );
+        }
+        setSelected(newSelected);
+    };
+
+    const isSelected = (id) => selected.indexOf(id) !== -1;
+
+    useEffect(() => {
+        setLoading(true)
+        const initialRows = [
+            {
```

Comment:
@Aiga115 do we have an endpoint for the organizations list already? Also, can we get the term (label) fromt the application context rather than from this hardcoded data? Thanks